phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1199,7 +1201,27 @@ private void filterIndexCheckerMutations(Map<TableInfo, List<Mutation>> mutation
             Entry<TableInfo, List<Mutation>> pair = mapIter.next();
             TableInfo tableInfo = pair.getKey();
             if (tableInfo.getOrigTableRef().getTable().isImmutableRows() && IndexUtil.isGlobalIndexCheckerEnabled(connection, tableInfo.getHTableName())) {
-                PTable table = PhoenixRuntime.getTable(connection, tableInfo.getHTableName().getString());
+                HTableDescriptor desc = null;
+                PTable table;
+                try {
+                    desc = connection.getQueryServices().getAdmin().getTableDescriptor(
```

Comment:
@swaroopak  This makes a remote call to the HMaster. Probably need to either cache table descriptors or find another way. Is the naming convention on the table not sufficient?

Reply:
What about MetaDataUtil.isViewIndex?